### ISSUE TITLE
.url.txt files don't appear to be working

### ISSUE THREAD

**Issue by kayila** [SKCraft/Launcher#127]
If both the .jar and the .jar.url.txt files in the mods folder, the mod gets pulled into the json file, and a copy of the jar gets put into the _upload folder along with it. However if I remove the jar and just have the .url.txt file it doesn't include the mod at all.

```
$ ls src/mods/_CLIENT/
CodeChickenCore-1.7.10-1.0.4.29-universal.jar
CodeChickenCore-1.7.10-1.0.4.29-universal.jar.url.txt
NotEnoughItems-1.7.10-1.0.3.74-universal.jar
NotEnoughItems-1.7.10-1.0.3.74-universal.jar.url.txt

$ cat src/mods/_CLIENT/*.txt
http://chickenbones.net/maven/codechicken/CodeChickenCore/1.7.10-1.0.4.29/CodeChickenCore-1.7.10-1.0.4.29-universal.jar
http://chickenbones.net/maven/codechicken/NotEnoughItems/1.7.10-1.0.3.74/NotEnoughItems-1.7.10-1.0.3.74-universal.jar

```

**Comment by kayila** [comment]
Upon doing some more testing... It's a windows bug. I just tried to build the same package on my linux box and it built without a problem and used the URLs to generate the download paths.

**Comment by sk89q** [comment]
The .jar should actually be required as the build tools generates the checksum from that .jar. The fact that it works on Linux without the .jar is strange.

**Comment by kayila** [comment]
Sorry, I was a little unclear. It works on linux with the jar file, but on windows it includes the jar file in the _upload folder instead of including just the link in the .json

**Comment by kayila** [comment]
It does not work on linux without the .jar

**Comment by dj3520** [comment]
So the actual issue here is that generating the modpack on Windows puts the file into the upload folder despite there being a url.txt, and linux does not do this.

**Comment by kayila** [comment]
Correct.

**Comment by TheQuantumGeneral** [comment]
Is it possible to set a mod up by URL only, so one could perhaps pull latest version from jenkins or similar?

**Comment by dj3520** [comment]
Needs to calculate hashes I believe.

**Comment by TheQuantumGeneral** [comment]
So probably not, as it would need to download the mod every time you launched to check if it had updated?

**Comment by HenryLoenwind** [comment]
I'd say the problem is this line:

https://github.com/SKCraft/Launcher/blob/master/creator-tools/src/main/java/com/skcraft/launcher/creator/Creator.java#L38 (origin: https://github.com/SKCraft/Launcher/commit/183fab764d1a72e864754f21268007eb14fe0163 )

It disables those URL overrides and they are never re-enabled anywhere.

**Comment by wizjany** [comment]
fixed #157
